oci8.go

```diff
@@ -389,8 +389,21 @@ func (drv *DriverStruct) Open(dsnString string) (driver.Conn, error) {
 		}
 		conn.svc = *svcCtxPP
 		doneLogon = true
+	}
+
+	// Create transaction context.
+	trans, _, err := conn.ociHandleAlloc(C.OCI_HTYPE_TRANS, 0)
+	if err != nil {
+		return nil, fmt.Errorf("allocate transaction handle error: %v", err)
+	}
 
+	// Set transaction context attribute of the service context.
+	err = conn.ociAttrSet(unsafe.Pointer(conn.svc), C.OCI_HTYPE_SVCCTX, *trans, 0, C.OCI_ATTR_TRANS)
+	if err != nil {
+		C.OCIHandleFree(*trans, C.OCI_HTYPE_TRANS)
```

Comment:
Move `C.OCIHandleFree(*trans, C.OCI_HTYPE_TRANS)` into `defer func(errP *error) {` just for future proofing. (Someone else might add something else later on after this line.)

Reply:
You mean deferring a custom function that receives the error but simply frees the handle? Or just deferring the call to `OCIHandleFree`?

Reply:
The deferring call above